lib/Sema/TypeCheckDecl.cpp

```diff
@@ -1051,10 +1051,8 @@ NeedsNewVTableEntryRequest::evaluate(Evaluator &evaluator,
                                      AbstractFunctionDecl *decl) const {
   auto *dc = decl->getDeclContext();
 
-  // FIXME: This is mysterious and seems wrong. However, changing it to return
-  // false (as it seems like it should) breaks a couple Serialization tests.
   if (!isa<ClassDecl>(dc))
-    return true;
+    return false;
```

Comment:
Can we assert here instead or is it too inconvenient to change the callers of this request?

Reply:
I think it's perfectly reasonable to call this on, say, extension methods.